src/components/Wizard/Wizard.tsx

```diff
@@ -0,0 +1,132 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+import './Wizard.scss';
+import { IPage } from './IPage';
+import { IStepProps } from './IStepProps';
+import { hashHistory } from 'react-router';
+import { Button } from '../../components/Button/Button';
+import {IButtonProps} from '../../components/Button/Button.Props';
+import {autobind } from '../../utilities/autobind';
+import Stepper from './Stepper';
+
+export interface IWizardProps {
+  steps?: Array<IStepProps>;
+  onPageRender?: (index: number) => JSX.Element;
+  onPageEnter?: (currentStepIndex: number, nextStepIndex: number) => void;
+  onPageLeave?: (currentStepIndex: number, nextStepIndex: number) => void;
+  onFinish?: () => void;
+  currentStep?: number;
+  currentPage?: IPage;
+  title?: string;
+  nextBtnState?: boolean;
+}
+
+export interface IWizardState {
+  currentStep?: number;
+}
+
+
+export default class Wizard extends React.Component<IWizardProps, IWizardState> {
+
+  constructor(props: IWizardProps) {
+    super(props);
+
+    this.state = {
+      currentStep: 0
+    };
+
+  }
+
+  componentWillMount() {
+    this.props.onPageEnter(0, 1);
+  }
+
+  private _nextStep(e) {
+    e.preventDefault();
+    if ((this.state.currentStep + 1) !== this.props.steps.length) {
+      this.setState({
+        currentStep: this.state.currentStep + 1
+      });
+      this.props.onPageRender((this.state.currentStep) + 1);
+      this.props.onPageLeave(this.state.currentStep, this.state.currentStep + 1);
+      this.props.onPageEnter(this.state.currentStep + 1, this.state.currentStep + 2);
+    }
+  }
+
+  private _backStep(e) {
+    e.preventDefault();
+    if (this.state.currentStep > 0) {
+      this.setState({
+        currentStep: this.state.currentStep - 1
+      });
+      this.props.onPageRender(this.state.currentStep - 1);
+      this.props.onPageLeave(this.state.currentStep, this.state.currentStep - 1);
+      this.props.onPageEnter(this.state.currentStep - 1, this.state.currentStep);
+    }
+  }
+
+  private _cancelCreateScript(e) {
+    e.preventDefault();
+    hashHistory.push('/');
```

Comment:
it would be better to use onCancel callback instead of directly using hashHistory